experiment/src/org/labkey/experiment/api/property/DomainPropertyImpl.java

```diff
@@ -777,6 +777,10 @@ else if (_pdOld != null)
                 }
             }
 
+            // Issue 44711: Prevent attachment and file field types from being converted to a different type
+            if ("file".equalsIgnoreCase(oldType.getInputType()) && oldType != newType)

```

Comment:
```suggestion
            if (PropertyType.FILE_LINK.getInputType().equalsIgnoreCase(oldType.getInputType()) && oldType != newType)

```

I haven't tested this yet but it would remove the magic text value.

Reply:
Thank you. I meant to go back and update that magic text value to use the const....but forgot. Done.